.github/workflows/codacy-analysis.yml

```
@@ -0,0 +1,46 @@
+# This workflow checks out code, performs a Codacy security scan
+# and integrates the results with the
+# GitHub Advanced Security code scanning feature.  For more information on
+# the Codacy security scan action usage and parameters, see
+# https://github.com/codacy/codacy-analysis-cli-action.
+# For more information on Codacy Analysis CLI in general, see
+# https://github.com/codacy/codacy-analysis-cli.
+
+name: Codacy Security Scan
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  codacy-security-scan:
+    name: Codacy Security Scan
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the repository to the GitHub Actions runner
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Execute Codacy Analysis CLI and generate a SARIF output with the security issues identified during the analysis
+      - name: Run Codacy Analysis CLI
+        uses: codacy/codacy-analysis-cli-action@1.1.0
```

The latest version is 4.1.0.
```suggestion
        uses: codacy/codacy-analysis-cli-action@master
```

---

Tghen lets use [4.1.0](https://github.com/codacy/codacy-analysis-cli-action/releases).

We can use dependabot to semi-automatically bump that, but I would not use a dev-branch for that.

---

```suggestion
        uses: codacy/codacy-analysis-cli-action@4.1.0
```

---

-> should also be done for Unicodify of course